### ISSUE TITLE
THIS LIBRARY IS DEAD / NO LONGER FUNCTIONS PROPERLY -- DO NOT USE!!

### ISSUE THREAD

**Issue by mihaibulic2** [priteshrnandgaonkar/react-native-call-detection#119]
**This library hasn't been updated in 4 years, and there are tons of open PRs and issues, I would suggest people steer clear of this library!**

Personally, I'm probably going to fork the library and leave it as is on iOS since it works just fine, and fix the permission stuff on Android.

**Summary of the issues I ran into so far on Android:**
- no types so hard to use in typescript
- readPhoneNumberAndroid must be true. 
    WHY: This library is really old, and back in the day the phone state permission was only
    needed if you wanted to read the caller's phone number; now it's needed
    for this listener to work at all. If this is not true, it skips the
    permission check which leads to a crash.
- if the user denies the permissions, you get a crash (surrounding call to the lib w/ try-catch does nothing)
- permission denied callback never fires (probably because of the crash)
- permissionMessage shows up as non-properly skinned dialog (ie if in dark mode title + bg are white on white)
- phone number never given even if you set readPhoneNumberAndroid (in [Android 11 they added a new permission](https://developer.android.com/about/versions/11/privacy/permissions#phone-numbers) for this, which I haven't tried, but I'm guessing this library doesn't request that permission, only READ_PHONE_STATE, so probably won't work out of the box)
- permission request to users is scary and wrong: READ_PHONE_STATE allows read only access to phone state, but the system popup when requesting the permission say "Allow <app> to make and manage phone calls", which is wrong.

- NOTE: seems to work fine on iOS

**Comment by ilies-space** [comment]
I totally agree, since the owner is not accepting PR's can we fork it, make the mentioned corrections and try to publish new corrected updated release ?

**Comment by abmaurya07** [comment]
Let's do that

**Comment by 0xbenon** [comment]
Thanks!

I'm using expo with managed workflow, so I had to create a expo-module, and I create a example app to test and it works with the new "TelephonyCallBack.CallStateListener" method from android and CXCallObserver from iOS:

https://github.com/benomzn/twilio-voice-exposdk51-example

The example is to basic, but maybe you can check the Kotlin code and write your own solution

**Comment by enriquebarco** [comment]
What other alternatives have you come across that provides this kind of functionality? I am looking for a simple way to detect call state - incoming, dialing, connected. And the phone number for Andriod. Please let me know!

**Comment by samueljim** [comment]
has anyone done any work on this? I need this library but i also want to update my code base

**Comment by Sonu7759** [comment]
has anyone done any work on this? I need this library but i also want to update my code base

**Comment by Sonu7759** [comment]
> What other alternatives have you come across that provides this kind of functionality? I am looking for a simple way to detect call state - incoming, dialing, connected. And the phone number for Andriod. Please let me know!

did you get the solution

**Comment by dgreasi** [comment]
Anyone found any alternatives?

**Comment by getsettalk** [comment]
please tell me another packages

**Comment by vivek0046** [comment]
Hey guys ,i have looked around 

there are no alternatives to this . @mihaibulic2 have you forked this ?

**Comment by NightScript370** [comment]
> NOTE: seems to work fine on iOS

Perhaps it is so functionality-wise. Nevertheless, importing this iOS library does not work on newer Expo versions. I have made a fork with this iOS fix, and I'd be happy to take on a maintenance job for this project should anyone want to submit Pull Requests, especially for the like of Android support:

https://github.com/NightScript370/react-native-call-detection

**Comment by dgreasi** [comment]
Probably this is the best alternative: [react-native-callkeep](https://github.com/react-native-webrtc/react-native-callkeep)

**Comment by LeXuanKhanh** [comment]
> > NOTE: seems to work fine on iOS
> 
> Perhaps it is so functionality-wise. Nevertheless, importing this iOS library does not work on newer Expo versions. I have made a fork with this iOS fix, and I'd be happy to take on a maintenance job for this project should anyone want to submit Pull Requests, especially for the like of Android support:
> 
> https://github.com/NightScript370/react-native-call-detection

hi can you make a fork public in npm, in case the author doesn't respond so we can continue to maintain the package

**Comment by LeXuanKhanh** [comment]
Nvm, for those who want to install the fork of @NightScript370, use this:
`yarn add react-native-call-detection@https://github.com/NightScript370/react-native-call-detection#patch-2`
However, this had an issue with Expo 53 due to deprecated `BatchedBridge.registerCallableModule`, if you want to use please downgrade your Expo and react native version.
